src/realm/query_engine.cpp

```diff
@@ -51,6 +51,9 @@ size_t ParentNode::find_first(size_t start, size_t end)
 
 void ParentNode::aggregate_local_prepare(Action TAction, DataType col_id, bool nullable)
 {
+    // Verify that the cached column accessor is still valid
+    verify_column(); // throws
```

Comment:
After looking at this in more detail, I don't think `aggregate_local_prepare` is the right place to call this. It happens too late in the process of query execution to catch all problems. `StringNode<Equal>::init`, for instance, accesses the column prior to `aggregate_local_prepare` being called. `ParentNode::init()` may be a better home for this check.

These two issues suggests that the tests added in this PR do not have sufficient coverage. I'd suggest looking at the extensive tests Thomas added for ensuring that [column insertions are handled correctly by queries](https://github.com/realm/realm-core/blob/854a2b5d8d310e3e1bdf0efc1ea420414ed0fbb8/test/test_query.cpp#L9527-L9836). The tests enumerate virtually all supported query operations, and similar tests would ensure complete coverage of the various query nodes for the column removal case as well.